src/iface/ethernet.rs

```diff
@@ -362,11 +365,15 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
             _ if handled_by_raw_socket =>
                 Ok(Packet::None),
 
-            _ => {
+            _ if eth_frame.dst_addr().is_unicast() && ipv4_repr.dst_addr.is_unicast() => {
+                // If the datagram does not have a payload greater than or equal
+                // to 64 bytes minus the size of the ipv4 header in length, only
+                // send back the size of given payload.
+                let payload_len = min(ip_payload.len(), SIZEOF_ICMP_ERR_PAYLOAD);
```

Comment:
According to RFC 792 up to 64 bytes of the original payload should be included. Since the header is separate, that is `64 - sizeof(ip header)`.